### PR TITLE
integrate loopback connector with strong-soap

### PR DESCRIPTION
**Pull request by deepakrkris**
strong-soap is a fork of node-soap with additional features
and broad coverage of soap specifications. loopback connector is being
integrated with strong-soap

### CODE REVIEW

**Comment by deepakrkris** [comment]
@slnode test please

**Comment by deepakrkris** [comment]
@slnode test please

**Comment by deepakrkris** [comment]
@slnode test please

**Comment by deepakrkris** [comment]
@slnode test please

**Comment by deepakrkris** [comment]
@slnode test please

**Comment by deepakrkris** [comment]
@slnode test please

**Comment by deepakrkris** [comment]
@slnode test please

**Comment by rmg** [comment]
@slnode test please

**Comment by deepakrkris** [comment]
@slnode test please

File: .gitignore.orig [hunk 1]
```diff
@@ -0,0 +1,13 @@
+.project
+.idea
+node_modules
+<<<<<<< e77f689d141df9c7e38a10fac365ab7a150b5372
```
Comment (raymondfeng) [inline .gitignore.orig]: This is apparently a git merging conflict. Please resolve it.

Comment (deepakrkris) [inline .gitignore.orig]: @raymondfeng it is not a required file , I will delete it

File: package.json [hunk 2]
```diff
@@ -4,13 +4,16 @@
   "description": "LoopBack SOAP Web Services Connector",
   "main": "index.js",
   "scripts": {
-    "test": "./node_modules/.bin/mocha -R spec --timeout 30000 test/*test.js"
+    "test": "./node_modules/.bin/mocha -R spec --timeout 30000 test/*test.js",
+    "pretest": "babel node_modules/strong-soap/src -d node_modules/strong-soap/lib"
```
Comment (raymondfeng) [inline package.json]: We should fix it in strong-soap instead.

File: package.json [hunk 3]
```diff
@@ -4,13 +4,16 @@
...
   },
   "dependencies": {
     "debug": "^2.1.1",
     "request": "^2.55.0",
-    "soap": "^0.16.0",
-    "strong-globalize": "^2.6.2"
+    "strong-soap": "latest",
```
Comment (raymondfeng) [inline package.json]: Please use a version range such as `"^1.0.0"`.

File: package.json [hunk 4]
```diff
@@ -4,13 +4,16 @@
...
+    "strong-globalize": "^2.6.2",
+    "babel-cli": "^6.9.0",
```
Comment (raymondfeng) [inline package.json]: Please remove the `babel-*` deps.

File: test/wsdls/stockquote.wsdl [hunk 5]
```diff
@@ -42,7 +42,7 @@
     </wsdl:portType>
 
     <wsdl:binding name="StockQuoteSoapBinding" type="tns:StockQuotePortType">
-        <soap:binding style="document" transport="http://schemas.xmlsoap.org/soap/http"/>
+        <soap:binding style="rpc" transport="http://schemas.xmlsoap.org/soap/http"/>
```
Comment (raymondfeng) [inline test/wsdls/stockquote.wsdl]: Is this change on purpose?

Comment (deepakrkris) [inline test/wsdls/stockquote.wsdl]: @raymondfeng yes, the end service is an rpc, but the wsdl wrongly had it as document

File: package.json (hunk 2 above)
Comment (deepakrkris) [inline package.json]: @raymondfeng , doing an npm install, actually brings the lib folder. it is only not working in the CI test environment

File: test/wsdls/stockquote.wsdl (hunk 5 above)
Comment (raymondfeng) [inline test/wsdls/stockquote.wsdl]: I think the WSDL is from http://www.webservicex.net/stockquote.asmx?WSDL and it's document style. Do you refer to https://github.com/strongloop/loopback-connector-soap/blob/master/test/security.test.js#L40? I cannot tell it's RPC though.

File: package.json (hunk 2 above)
Comment (raymondfeng) [inline package.json]: CI uses its own staging npm registry. We need to make sure babel is run before `npm publish`.

Comment (deepakrkris) [inline package.json]: @raymondfeng babel is added to prepublish

File: package.json (hunk 3 above)
Comment (deepakrkris) [inline package.json]: @raymondfeng done

File: package.json (hunk 4 above)
Comment (deepakrkris) [inline package.json]: @raymondfeng done

File: test/wsdls/stockquote.wsdl (hunk 5 above)
Comment (deepakrkris) [inline test/wsdls/stockquote.wsdl]: i have changed that back to "document"
